### ISSUE TITLE
Warning: a promise was created in a handler, but was not returned from it

### ISSUE THREAD

**Issue by Sicria** [OptimalBits/bull#763]
Seems like this error is still occurring from https://github.com/OptimalBits/bull/issues/734

``Warning: a promise was created in a handler at node_modules/bull/lib/queue.js:313:17 but was not returned from it``

**Comment by ritter** [comment]
I can confirm I am seeing this one as well.

`Warning: a promise was created in a handler at node_modules/bull/lib/queue.js:313:17 but was not returned from it, see http://goo.gl/rRqMUw`

**Comment by jeffijoe** [comment]
I am also seeing this warning.

**Comment by cristianfd** [comment]
I'm seeing this warning in .getJob line (variables are ok):
```
queue.getJob(job.id).then((job2) => {
....
});
```

**Comment by albertyou2** [comment]
hi @cristianfd 
 have you sovle this problem?

**Comment by cristianfd** [comment]
TLDR; No
But:
I dont know if is a bull issue or and issue with 'our' code, so my solution was to take other way and it worked.
I solve all my problems using [separate process](https://github.com/OptimalBits/bull#separate-processes). so I recommend to first review your own code and do a try with separate process, the promise mode gives you more control.

**Comment by manast** [comment]
it is still a Warning, not a dangerous one besides, although always nice to have warning free code
